File: app/src/main/java/com/jat/medilinkapp/util/SharePreferencesUtil.java
package com.jat.medilinkapp.util;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class SharePreferencesUtil {
  private Context context;

  public SharePreferencesUtil(Context context) {
    this.context = context;
  }

  public void setValue(String key, String value) {
    SharedPreferences.Editor edit = PreferenceManager.getDefaultSharedPreferences(context).edit();
    edit.putString(key, value);
    edit.apply();
  }

  public void setValue(String key, boolean value) {
    SharedPreferences.Editor edit = PreferenceManager.getDefaultSharedPreferences(context).edit();
    edit.putBoolean(key, value);
    edit.apply();
  }

  public String getValue(String key, String defaulValue) {
    SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    String value = sharedPreferences.getString(key, defaulValue);
    return value;
  }

  public boolean getValue(String key, boolean defaulValue) {
    SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    boolean value = sharedPreferences.getBoolean(key, defaulValue);
    return value;
  }
}

File: app/src/main/java/com/jat/medilinkapp/util/SharePreferencesUtil.kt
package com.jat.medilinkapp.util

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class SharePreferencesUtil(private val context: Context) {

    fun setValue(key: String, value: String) {
        val edit = PreferenceManager.getDefaultSharedPreferences(context).edit()
        edit.putString(key, value)
        edit.apply()
    }

    fun setValue(key: String, value: Boolean) {
        val edit = PreferenceManager.getDefaultSharedPreferences(context).edit()
        edit.putBoolean(key, value)
        edit.apply()
    }

    fun getValue(key: String, defaulValue: String): String? {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        return sharedPreferences.getString(key, defaulValue)
    }

    fun getValue(key: String, defaulValue: Boolean): Boolean {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        return sharedPreferences.getBoolean(key, defaulValue)
    }
}
